test/framework/options.py

```diff
@@ -5929,6 +5928,15 @@ def test_sanity_check_only(self):
         outtxt = self.eb_main(args + ['--skip-extensions'], do_build=True, raise_error=True)
         self.assertTrue("Sanity check for toy successful" in outtxt)
 
+        # restore fail, we want a passing sanity check for the next check
+        move_file(libbarbar + '.moved', libbarbar)
+
+        # check use of --sanity-check-only when installation directory is read-only;
+        # cfr. https://github.com/easybuilders/easybuild-framework/issues/3757
+        adjust_permissions(ebroottoy, stat.S_IWUSR, add=False, recursive=True)
```

Comment:
Unless guaranteed to be so already, you need to remove group (and other) write permissions too.

Reply:
OK, doesn't hurt to be a bit more strict here...